### PR TITLE
Issue175 use xlparser to parse print area

### PR DESCRIPTION
**Pull request by andrewkress**
#175 is the main issue addressed.

Bigger changes include removing the .NET 3.5 project build and updating the minimum .NET version to 4.5.

These changes were required to use the XLParser Library to allow more accurate parsing of formulas and area references.

### CODE REVIEW

File: ClosedXML/Excel/XLWorkbook_Load.cs [hunk 1]
```diff
@@ -933,7 +933,7 @@ private static bool IsRowReference(string sheetArea)
         private static void ParseReference(string item, out List<Tuple<string, string>> sheetNameAndArea) 
         {
             sheetNameAndArea = new List<Tuple<string, string>>();
-            ParseTreeNode nodes = ExcelFormulaParser.Parse("Test(" + item + ")");
+            ParseTreeNode nodes = ExcelFormulaParser.Parse(String.Format("DummyFormula({0})", item));
```
Comment (andrewkress) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: For the library to parse the print area, it had to be mocked up in a formula format.  Once parsed, the information is dropped.

**Comment by igitur** [comment]
This is good work, but there are some issues to consider. Dropping of .NET 3.5 is potentially problematic. I don't have metrics of the number of users of the release, but recently someone commented on another issue and I realised he was using .NET 3.5.

Also, .NET Standard support is a often requested feature, and XLParser and its underlying Irony will first have to support .NET Standard. I did ask about it at https://irony.codeplex.com/discussions/661482 , but the project looks pretty dead. The last commit was in 2013 (I didn't realise that earlier) and it makes me hesitant to have it as a dependency if the library isn't actively developed. I've asked them whether they plan to support .NET Standard at https://irony.codeplex.com/discussions/661482

File: ClosedXML/Excel/XLWorkbook_Load.cs (hunk 1 above)
Comment (igitur) [inline ClosedXML/Excel/XLWorkbook_Load.cs]: Yes, I think you can just wrap it in `(` and `)` too, if I remember correctly.

**Comment by andrewkress** [comment]
Yes, those are both good points and need to be considered if this is to get merged.  Using these libraries becomes a nightmare if you need to still support .NET 3.5.  
At almost 10 years old, are there situations where someone does not have a newer version of .NET installed?
As for support of .NET Standard, we can wait and see.  It appears that the maintainer is still active in the discussion there, but it doesn't appear that much has changed in the codebase in years.

**Comment by andrewkress** [comment]
Closing this, didn't realize it was still open. The issue was resolved with a different pull request.

**Pull request closed without merging**